lib/puppet/provider/package/pip.rb

```diff
@@ -56,11 +56,8 @@ def query
   # cache of PyPI's package list so this operation will always have to
   # ask the web service.
   def latest
-    client = XMLRPC::Client.new2("http://pypi.python.org/pypi")
-    client.http_header_extra = {"Content-Type" => "text/xml"}
-    client.timeout = 10
-    result = client.call("package_releases", @resource[:name])
-    result.first
+    result = JSON.parse(open("https://pypi.python.org/pypi/#{@resource[:name]}/json").read)
```

Comment:
I filed https://tickets.puppetlabs.com/browse/PUP-5069 to update the HTTP API. In the meantime, I agree with @branan that following the [PMT example](https://github.com/puppetlabs/puppet/blob/master/lib/puppet/forge/repository.rb#L135-L142) might be the way to go for now. It avoids the issue because it creates its own `Net::HTTP` objects and calls `OpenSSL::X509::Store#set_default_paths` directly. You'll also want to apply the http connect and read timeouts.